### ISSUE TITLE
Lsyncd does not create a directory with the current date if you specify /dir/`date +%Y-%m-%d`

### ISSUE THREAD

**Issue by xpom43** [lsyncd/lsyncd#665]
Hello, lsyncd does not create a directory with the current date if you specify /dir/`date +%Y-%m-%d` in backup_dir=/dir/`date +%Y-%m-%d`, but creates a folder "` date +%Y-%m-%d`" Rsync creates if you specify --backup-dir=/dir/`date +%Y-%m-%d` a directory named "/dir/2022-03-23" . How can lsyncd create a directory of deleted files with the current date?

**Comment by julianuu** [comment]
It doesn't seem so, the only info I found in that direction is the advice here https://github.com/lsyncd/lsyncd/issues/433 to code it yourself in "layer 1".

I'm interested in that feature as well so I'm happy to help, but I don't even know where to start. It seems that the corresponding parameter is set here https://github.com/lsyncd/lsyncd/blob/master/default-rsync.lua#L678, and then `rsync` is called with that parameter here https://github.com/lsyncd/lsyncd/blob/master/default-rsync.lua#L209. So I guess one would somehow have to edit that `run_action` function, but I don't know how to do that via config files.

**Comment by julianuu** [comment]
Yeah so that indeed worked. I made sure `rsync` is always called with `--suffix=…` and instead of `--backup-dir=…`, but that should work similarly.

One can create a copy `default.rsync`, give it a different name and edit all the instances of `spawn(…)` so that `rsync` is always called with `"--suffix=_" .. os.date("%F_%T)`. Those occur in the functions `run_action` and `rsync.full`, so we have to rewrite only those.

Here is how my config looks like
```
local rsync_datesuffix = { default.rsync }

local run_action = function
	(
		inlet,
		elist
	)
…
```
Then comes the rest of the function `run_action` from https://github.com/lsyncd/lsyncd/blob/master/default-rsync.lua however with every instance of `rsync.` replaced by `rsync_datesuffix_.`
Except, the call of `spawn(…)` is replaced by
```
	local date = os.date("%F_%T")

	spawn(
		elist,
		config.rsync_datesuffix.binary,
		'<', table.concat( filterI, '\000' ),
		config.rsync_datesuffix._computed,
		'--suffix=_' .. date,
		'-r',
		delete,
		'--force',
		'--from0',
		'--include-from=-',
		'--exclude=*',
		config.source,
		target
	)
```

Then we do the same thing with the function `full`, i.e.
```
rsync_datesuffix.full = function
	(
		event
	)
…
```
and again the rest of the function with `rsync.` replaced by `rsync_datesuffix.`. Here `spawn(…)` is occurs in several places, so we define `local date = os.date("%F_%T")` before those, and add the argument `'--suffix=_' .. date` after `config.rsync_datesuffix._computed` to each call of `spawn(…)`.

Next we write the rest of the config like
```
settings {
    …
}

sync {
    rsync_datesuffix,
    source="…",
    …
}
```

Not sure if this is the best way to do it but it seems to work.
